### ISSUE TITLE
Distributed DDL ALTERs not working on ReplicatedMergeTree?

### ISSUE THREAD

**Issue by jpiper** [yandex/ClickHouse#2207]
(version 1.1.54358)

Are ALTER queries on ReplicatedMergeTree not supported via distributed DDL?

When I try to do something like this on a ReplicatedMergeTree

```
alter table x on cluster y drop partition z;
```

(or if I try to add a column)

I end up with the error (on all nodes)

> trying: Table logs_shard is replicated, but shard #9 isn't replicated according to its cluster definition

However, everything appears to be setup correctly - all the data are correctly replicated and queries are working fine, so I don't know what is wrong in this case.

```xml
<shard>
    <replica>
        <host>NODE1</host>
        <port>9000</port>
        <user>user</user>
        <password>password</password>
    </replica>
    <replica>
        <host>NODE2</host>
        <port>9000</port>
        <user>user</user>
        <password>password</password>
    </replica>
</shard>
```

```xml
    <yandex>
        <macros>
            <shard>04</shard>
            <replica>NODE1</replica>
        </macros>
    </yandex>
...
    <yandex>
        <macros>
            <shard>04</shard>
            <replica>NODE2</replica>
        </macros>
    </yandex>
```

And the DDL looks something like
```sql
CREATE TABLE db1.table_shard on cluster cluster (
date Date,
site String,
value Float64)
ENGINE = ReplicatedMergeTree('/clickhouse/tables/db1/table_shard/{shard}/', '{replica}')
PARTITION BY toString(date)
ORDER BY (date, site) SETTINGS index_granularity = 8192
```

**Comment by ludv1x** [comment]
You should add <internal_replication>true</internal_replication> to the cluster definition. Like that:
https://github.com/yandex/ClickHouse/blob/bf1c4d156a72f915a78428dbc8c5493eaba67db2/dbms/tests/integration/test_distributed_ddl/configs/config.d/clusters.xml#L7

**Comment by jpiper** [comment]
@ludv1x Thanks for the reply! So now I get a different error, but only on some nodes...

> Code: 164, e.displayText() = DB::Exception: Received from XXXXXXXX:9000, 10.110.208.170. DB::Exception: Cannot execute query in readonly mode., e.what() = DB::Exception

Which is odd as I am logged in as a user with admin permissions

**Comment by filimonov** [comment]
Didn't you see also the following message?

https://github.com/yandex/ClickHouse/blob/08170d0d779b12eff05777e12f3d6b086e54ae02/dbms/src/Interpreters/DDLWorker.cpp#L239

Check that part of your config.xml and your profile:
```sql
    <distributed_ddl>
        <!-- Path in ZooKeeper to queue with DDL queries -->
        <path>/clickhouse/task_queue/ddl</path>

        <!-- Settings from this profile will be used to execute DDL queries -->
        <!-- <profile>default</profile> -->
    </distributed_ddl>
```

**Comment by jpiper** [comment]
My config.xml looks like that already.

I did not see that message - some nodes gave the "Cannot execute query in readonly mode" - others did not (it looked successful).

My default user is read-only and I set the admin username and password in `<remote_servers>` I am logged in via the admin user when I run the command. Could this be fixed in #2202?
